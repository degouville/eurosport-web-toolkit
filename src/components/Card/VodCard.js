import React from 'react';
import PropTypes from 'prop-types';
import { Card, CardHeader, CardContent, CardFooter } from './base';

import Heading from '../../elements/Heading';
import Text from '../../elements/Text';
import ChannelLogo from '../../elements/ChannelLogo';
import Icon from '../../elements/Icon';

import styled, { css } from 'react-emotion';

const base = css`
  color: goldenrod;
`;

const StyledHeading = styled(Heading)`
  ${base}
  text-transform: uppercase;
  font-size: 1em;
  letter-spacing: 0.2em;
`;

const ContentDescription = styled(Text)`
  font-weight: bold;
`;

const StyledCardHeader = styled(CardHeader)`
  position: relative
`;

const StyledLiveLabel = styled.div`
  background-color: red;
  color: white;
  position: absolute;
  top: 10px;
  right: 10px;
`;

const StyledPlayIcon = styled(Icon)`
  position: absolute;
  bottom: 10px;
  left: 10px;
`;

const VodCard = ( {live = false} ) => (
  <Card>
    <CardHeader style={{position: 'relative'}}>
      <img src="https://i.eurosport.com/2018/10/29/2450727-50913270-2560-1440.jpg?w=200" />
      <StyledPlayIcon type='play' width='60' />
      {live && <StyledLiveLabel>Live</StyledLiveLabel>}
    </CardHeader>
    <CardContent>
      <StyledHeading as="h1">Heading</StyledHeading>
      <ContentDescription>Lorem ipsum dolor sit.</ContentDescription>
    </CardContent>
    <CardFooter>
      <ChannelLogo />
      <Text>15:00 - 16:00</Text>
    </CardFooter>


  </Card>);

VodCard.displayName = 'VodCard';

VodCard.propTypes = {
  live: PropTypes.bool.isRequired,
};

export default VodCard;
