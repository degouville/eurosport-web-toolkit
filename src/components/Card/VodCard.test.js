import React from 'react';
import { shallow } from 'enzyme';
import VodCard from './VodCard';

it('renders a VodCard', () => {
  expect(shallow(<VodCard />)).toMatchSnapshot();
});
