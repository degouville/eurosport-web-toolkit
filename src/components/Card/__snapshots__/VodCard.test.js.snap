// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vod card Renders a Vod card 1`] = `
<CompactCard
  card={
    Object {
      "category": "Tennis",
      "description": "Description",
      "img": "https://i.eurosport.com/2018/10/29/2450727-50913270-2560-1440.jpg?w=200",
      "timestamp": "01:09:28",
      "title": "Youth olympic summer games",
    }
  }
  icon={
    <img
      alt="play"
      height={20}
      src="play.svg"
    />
  }
/>
`;
